wildlifelicensing/apps/applications/views/entry.py

```diff
@@ -184,23 +189,78 @@ def get(self, request, *args, **kwargs):
                 messages.error(self.request, e.message)
                 return redirect('wl_applications:new_application')
 
-            application.licence_type = WildlifeLicenceType.objects.get(code_slug=self.args[0])
+            application.licence_type = WildlifeLicenceType.objects.get(id=self.args[0])
+
+            application.variants.clear()
+
+            for index, variant_id in enumerate(request.GET.getlist('variants', [])):
+                try:
+                    variant = Variant.objects.get(id=variant_id)
+
+                    ApplicationVariantLink.objects.create(application=application, variant=variant, order=index)
+                except Variant.DoesNotExist:
+                    pass
+
             application.save()
 
             return redirect('wl_applications:check_identification')
 
-        categories = {}
+        categories = []
+
+        def _get_variants(variant_group, licence_type, current_params):
+            variants = []
+
+            for variant in variant_group.variants.all():
+                variant_dict = {'text': variant.name}
+
+                if variant_group.child is not None:
+                    variant_dict['nodes'] = _get_variants(variant_group.child, licence_type, current_params + [variant.id])
```

Comment:
When I'll be Tech Lead, I will impose terminal cases first. For now it's okay :-)